docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,1385 @@
+- The `pages` reserved node holds the navigation items.
+- The navigation items are defined within a logical node (`configurable-bundle-templates`) according to business requirements.
+- The `bundle` identifies in which `module` the processing `controller` is located.
+```xml
+<?xml version="1.0"?>
+<config>
+  <product>
+    <pages>
+      <configurable-bundle-templates>
+        <label>Configurable Bundle Templates</label>
+        <title>Configurable Bundle Templates</title>
+        <bundle>configurable-bundle-gui</bundle>
+        <controller>template</controller>
+        <action>index</action>
+        <visible>1</visible>
+        <pages>
+          <configurable-bundle-template-create>
+            <label>Create Configurable Bundle Template</label>
+            <title>Create Configurable Bundle Template</title>
+            <bundle>configurable-bundle-gui</bundle>
+            <controller>template</controller>
+            <action>create</action>
+            <visible>0</visible>
+          </configurable-bundle-template-create>
+        </pages>
+      </configurable-bundle-templates>
+    </pages>
+  </product>
+</config>
+```
+
+## Permission Plugin
+
+**Description**
+
+Permission plugins are a way to put a scope on the usage of an application during a request lifecycle.
+
+**Conventions**
+
+- Permission plugins MUST implement `\Spryker\Shared\PermissionExtension\Dependency\Plugin\PermissionPluginInterface`.
+
+**Guidelines**
+- When using one of the traits in a model, use a protected constant string to reference the permission plugin that will be used in the model. As an annotation for this string you should add `@uses` annotation to the `PermissionPlugin::KEY` string that you will compare against (see in example below).
+- The `\Spryker\[Application]\Kernel\PermissionAwareTrait` trait allows the model in the corresponding application to check if a permission is granted to an application user.
+
+**Example**
+
+```php
+class QuotePermissionChecker implements QuotePermissionCheckerInterface
+{
+use PermissionAwareTrait;
+
+    /**
+     * @uses \Spryker\Client\SharedCart\Plugin\ReadSharedCartPermissionPlugin::KEY
+     */
+    protected const PERMISSION_PLUGIN_KEY_READ_SHARED_CART = 'ReadSharedCartPermissionPlugin';
+
+    public function checkQuoteReadPermission(QuoteTransfer $quoteTransfer): bool
+    {
+        ...
+      return $this->can(
+        static::PERMISSION_PLUGIN_KEY_READ_SHARED_CART,
+        $quoteTransfer->getCustomer()->getCompanyUserTransfer()->getIdCompanyUser(),
+        $quoteTransfer->getIdQuote(),
+      );
+    }
+```
+
+## Persistence Schema
+
+**Description**
+
+The schema file defines the module's tables and columns (see [Propel Documentation - Schema XML](#https://propelorm.org/documentation/reference/schema.html)).
+
+**Conventions**
+
+- Table name MUST follow the format `spy_[module]_[domain entity]` (eg: `spy_customer_address`, `spy_customer_address_book`).
```

Comment:
IMO we should outline the naming format for the file as well e.g. `spy_[module].schema.xml (everything lowercase, if module name is CamelCase use and dashed e.g. product-url)`